### PR TITLE
Add in add topic method to app and display it onto each section

### PR DESCRIPTION
**Pull request by collinkallery**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling (Styling to new or existing files)

### Detailed Description
- Add in method for adding a topic to each of the homepage stories 
- Shows the topic before each of the homepage stories 
- Resets state after the last fetch 
### Why is this change required? What problem does it solve?
- Further clarification for the user to know what type of articles they are looking at. 
### Was any previously written code changed?
- No 
### Directory modified:

- [x] - App
- [ ] - Article
- [x] - ArticleContainer
- [x] - ArticleExpanded
- [ ] - ArticlePreview
- [ ] - Comment
- [ ] - CommentContainer
- [ ] - Login
- [ ] - NavBar
- [ ] - UserInfo
- [ ] - WelcomePage

### CODE REVIEW

File: src/Components/App/App.js [hunk 1]
```diff
@@ -18,37 +18,52 @@ class App extends Component {
       bestStoryIDs: [],
       topStoryIDs: [],
       homePageStories: [],
+      test: ''
```
Comment (collinkallery) [inline src/Components/App/App.js:21]: Will need to remove this

File: src/Components/App/App.js [hunk 2]
```diff
@@ -18,37 +18,52 @@ class App extends Component {
...
+  };
+
+  addTopic = (story, topic) => {
+    const matchingStory = this.state.homePageStories.find(specificStory => specificStory.id == story.id)
+    matchingStory['topic'] = topic;
+    const index = this.state.homePageStories.indexOf(matchingStory);
+    this.state.homePageStories.splice(index, 1);
+    this.setState({
+      homePageStories: [...this.state.homePageStories, matchingStory]
+    })
```
Comment (collinkallery) [inline src/Components/App/App.js:58]: Finds the story and adds the correct topic

File: src/Components/ArticlePreview/ArticlePreview.js [hunk 3]
```diff
@@ -2,9 +2,12 @@ import React, {useState} from 'react';
...
 import styled from 'styled-components'
 import {fetchImage} from '../../apiCalls'
+import {darkTheme} from "../../theme/globalStyle";
 
-const StoryCard = styled.div`
-  background-color: purple;
+const {primaryPurple, primaryBlue, secondaryTeal, background, textColor, error} = darkTheme;
+
+const Wrapper = styled.div`
+  background-color: ${primaryPurple};
```
Comment (collinkallery) [inline src/Components/ArticlePreview/ArticlePreview.js:10]: Import of dark theme

File: src/Components/App/App.js (hunk 2 above)
Comment (laurenlucero) [reply src/Components/App/App.js:58]: teamwork makes the dreamwork.

**Pull request merged**